operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -610,11 +611,24 @@ protected V1ObjectMeta createMetadata() {
         .putLabelsItem(
             LabelConstants.SERVERRESTARTVERSION_LABEL, getServerSpec().getServerRestartVersion());
 
+    if (!getDomain().isUseOnlineUpdate()) {
+      Optional.ofNullable(miiDomainZipHash)
+            .ifPresent(hash -> addHashLabel(metadata, LabelConstants.MODEL_IN_IMAGE_DOMAINZIP_HASH, hash));
+    }
     Optional.ofNullable(miiModelSecretsHash)
           .ifPresent(hash -> addHashLabel(metadata, LabelConstants.MODEL_IN_IMAGE_MODEL_SECRETS_HASH, hash));
+
+    // Add legacy prometheus annotations. These have to remain, as they are included in the computation
+    // of the pod sha256 hash, and removing them will cause pods to roll when customers upgrade to new
+    // versions of the operator.
+    AnnotationHelper.annotateForPrometheus(metadata, "/wls-exporter", getMetricsPort());
```

Comment:
This line may cause a NPE on unboxing a null Integer if both getListenPort() and getSslPort() is null in getMetricsPort().

Reply:
True - of course, such a domain would be pretty worthless. IMO the logic is wrong in any case, but I didn't want to change it here - all I have done is refactored the existing logic.

Reply:
Operator can definitely hit this problem without Johnny's fix in PR #2256 with secure mode enabled but SSL is not configured. Johnny's changes tire up the loose ends in some of the configuration permutations. I am not sure if there will be any configuration that may hit this even with PR #2256.

Reply:
The ugly part of this NPE is that when it happens it is hard to figure out what is going on. It would be helpful if we could at least detect the situation and log a warning message, instead of letting NPE occur.

Reply:
I've added a change that should catch that.